File: tests/agent_workloads/adapters/repository_marker_scan.mjs
import { readdirSync, readFileSync } from "node:fs";
import { join, relative, sep } from "node:path";

if (process.argv.length !== 3) {
  throw new Error("usage: repository_marker_scan INPUT_ROOT");
}

const root = process.argv[2];
const files = [];
function walk(dir) {
  for (const entry of readdirSync(dir, { withFileTypes: true })) {
    const path = join(dir, entry.name);
    if (entry.isDirectory()) walk(path);
    else if (entry.isFile()) files.push(path);
  }
}

walk(root);
const rows = [];
for (const file of files) {
  const count = readFileSync(file, "utf8")
    .split(/\r?\n/)
    .filter((line) => line.includes("agent_workload:")).length;
  if (count) {
    rows.push(`/${relative(root, file).split(sep).join("/")}|${count}`);
  }
}
console.log(rows.sort().join("\n"));

File: tests/agent_workloads/adapters/repository_marker_scan.py
import sys
from pathlib import Path


if len(sys.argv) != 2:
    raise SystemExit("usage: repository_marker_scan INPUT_ROOT")

root = Path(sys.argv[1])
rows = []
for file in root.rglob("*"):
    if not file.is_file():
        continue
    count = sum("agent_workload:" in line for line in file.read_text().splitlines())
    if count:
        rows.append(f"/{file.relative_to(root).as_posix()}|{count}")

print(*sorted(rows), sep="\n")
